# Pandas Project 2

Netflix Movies Analysis

In [82]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("shivamb/netflix-shows")

print("Path to dataset files:", path)

Using Colab cache for faster access to the 'netflix-shows' dataset.
Path to dataset files: /kaggle/input/netflix-shows


In [83]:
# Getting the data
import pandas as pd
data = pd.read_csv('/kaggle/input/netflix-shows/netflix_titles.csv')

In [84]:
# Converting the data to a pandas dataframe
df = pd.DataFrame(data)
df.head()

,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,NaN,United States,"September 25, 2021",2020,PG-13,90 min,Documentaries,"As her father nears the end of his life, filmm..."
1,s2,TV Show,Blood & Water,NaN,"Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...",South Africa,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, TV Dramas, TV Mysteries","After crossing paths at a party, a Cape Town t..."
2,s3,TV Show,Ganglands,Julien Leclercq,"Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act...",To protect his family from a powerful drug lor...
3,s4,TV Show,Jailbirds New Orleans,NaN,NaN,NaN,"September 24, 2021",2021,TV-MA,1 Season,"Docuseries, Reality TV","Feuds, flirtations and toilet talk go down amo..."
4,s5,TV Show,Kota Factory,NaN,"Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...",India,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, Romantic TV Shows, TV ...",In a city of coaching centers known to train I...


In [85]:
# Checking the null values
df.isnull().sum()

,0
show_id,0
type,0
title,0
director,2634
cast,825
country,831
date_added,10
release_year,0
rating,4
duration,3


In [86]:
# Exploring the data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


In [87]:
# Dropping the null values because we have a very large dataset
df.dropna(inplace=True)

In [88]:
# Again confirming the dataframe doesn't contains any null values
df.isnull().sum()

,0
show_id,0
type,0
title,0
director,0
cast,0
country,0
date_added,0
release_year,0
rating,0
duration,0


In [89]:
# Checking the shape of the dataframe
df.shape

(5332, 12)

In [112]:
# Removing leading whitespaces
columns_to_strip = ['type', 'title', 'director', 'cast', 'country', 'rating', 'duration', 'listed_in', 'description']
for column in columns_to_strip:
  df[column] = df[column].str.strip()

## Content Composition

Count of Movies vs TV Shows

In [113]:
count_mov_tv = df["type"].value_counts()
count_mov_tv

,count
type,
Movie,5185
TV Show,147


Distribution of titles across release years

In [114]:
dist_title_by_year = df.groupby("release_year")["title"].value_counts()
dist_title_by_year

release_year  title                  
1942          The Battle of Midway       1
1944          Tunisian Victory           1
1945          Know Your Enemy - Japan    1
1946          Let There Be Light         1
1947          Thunderbolt                1
                                        ..
2021          Wish Dragon                1
              Wish You                   1
              Xico's Journey             1
              Xtreme                     1
              YES DAY                    1
Name: count, Length: 5331, dtype: int64

Titles added per year

In [115]:
titles_per_year = df.groupby("release_year")["title"].count().sort_values(ascending=False)
titles_per_year

,title
release_year,
2017,657
2018,648
2016,577
2019,519
2020,442
...,...
1946,1
1963,1
1961,1


## Geographical Insights

Top 10 countries producing content

In [122]:
top_10_countries = df["country"].str.strip().value_counts().sort_values(ascending=False)
top_10_countries.head(10)

,count
country,
United States,1846
India,875
United Kingdom,183
Canada,107
Spain,91
Egypt,90
Nigeria,88
Japan,83
Turkey,79


Titles by country

In [130]:
df_exploded = df.assign(country=df['country'].str.split(',')).explode('country')
df_exploded['country'] = df_exploded['country'].str.strip()
titles_by_country = df_exploded.groupby("country")["title"].count().sort_values(ascending=False)
titles_by_country

,title
country,
United States,2485
India,940
United Kingdom,485
Canada,295
France,293
...,...
Senegal,1
Sri Lanka,1
Sudan,1


Compare diversity of content by region

In [133]:
diversity_by_region = df_exploded.groupby("country")["type"].nunique().sort_values(ascending=False)
diversity_by_region

,type
country,
Argentina,2
Australia,2
Belgium,2
Brazil,2
India,2
...,...
Vatican City,1
Venezuela,1
Vietnam,1


## Genre & Rating Analysis

Most common genres

In [121]:
common_genre = df["listed_in"].str.split(',').explode().str.strip().value_counts().sort_values(ascending=False)
common_genre.head(15)

,count
listed_in,
International Movies,2369
Dramas,2293
Comedies,1553
Action & Adventure,806
Independent Movies,740
Romantic Movies,579
Thrillers,547
Children & Family Movies,503
Documentaries,391


Most common TV ratings

In [98]:
common_ratings = df['rating'].value_counts().sort_values(ascending=False)
common_ratings

,count
rating,
TV-MA,1822
TV-14,1214
R,778
PG-13,470
TV-PG,431
PG,275
TV-G,84
TV-Y7,76
TV-Y,76


Titles by rating category

In [99]:
titles_by_rating = df.groupby("rating")["title"].count().sort_values(ascending=False)
titles_by_rating

,title
rating,
TV-MA,1822
TV-14,1214
R,778
PG-13,470
TV-PG,431
PG,275
TV-G,84
TV-Y,76
TV-Y7,76


## Release Trends

Titles released over time

In [100]:
titles_over_time = df["release_year"].value_counts().sort_values(ascending=False)
titles_over_time

,count
release_year,
2017,657
2018,648
2016,577
2019,519
2020,442
...,...
1945,1
1946,1
1942,1


## Director & Cast Insights

Directors with the most titles

In [136]:
dir_with_most_titles = df["director"].str.split(',').explode().str.strip().value_counts().sort_values(ascending=False)
dir_with_most_titles

,count
director,
Jan Suter,21
Raúl Campos,19
Marcus Raboy,15
Jay Karas,15
Cathy Garcia-Molina,13
...,...
Steve Race,1
Brett Harvey,1
Gabe Klinger,1


Cast members with most appearances

In [137]:
cast_with_most_app = df['cast'].str.split(',').explode().str.strip().value_counts().sort_values(ascending=False)
cast_with_most_app

,count
cast,
Anupam Kher,42
Shah Rukh Khan,34
Naseeruddin Shah,31
Akshay Kumar,30
Om Puri,30
...,...
Carolyn Bridget Kennedy,1
Krissiri Sukhsvasti,1
Atchareeya Potipipittanakorn,1


Most frequent genre combinations

In [140]:
genre_combination = df['listed_in'].str.split(',').explode().str.strip().value_counts().sort_values(ascending=False)
genre_combination.head(15)

,count
listed_in,
International Movies,2369
Dramas,2293
Comedies,1553
Action & Adventure,806
Independent Movies,740
Romantic Movies,579
Thrillers,547
Children & Family Movies,503
Documentaries,391


## Summary: Data Analysis Key Findings

### Content Composition Insights

*   **Count of Movies vs. TV Shows**: The dataset contains significantly more Movies than TV Shows. Movies account for `5185` titles, while TV Shows have `147` titles.
*   **Distribution of Titles Across Release Years**: The majority of content was released in recent years. The year `2017` had the most titles released (`657`), followed by `2018` (`648`) and `2016` (`577`). Content spans from as early as `1942` to `2021`.

### Geographical Insights

*   **Top 10 Countries Producing Content**: The United States dominates content production, followed by India.
    *   United States: `2485` titles
    *   India: `940` titles
    *   United Kingdom: `485` titles
    *   Canada: `295` titles
    *   France: `293` titles
    *   Spain: `192` titles
    *   Germany: `186` titles
    *   Japan: `170` titles
    *   Mexico: `125` titles
    *   Egypt: `119` titles
*   **Diversity of Content by Region (Movies and TV Shows)**: Many countries produce both Movies and TV Shows (indicated by `nunique()` type count of 2). This indicates a diverse content offering. Countries like Argentina, Australia, Belgium, Brazil, and India all produce both content types, showcasing a varied production landscape.

### Genre & Rating Analysis Insights

*   **Most Common Genres**: "International Movies" and "Dramas" are the most prevalent genres.
    *   International Movies: `2369`
    *   Dramas: `2293`
    *   Comedies: `1553`
    *   Action & Adventure: `806`
    *   Independent Movies: `740`
*   **Most Common TV Ratings / Titles by Rating Category**: "TV-MA" is the most frequent rating, followed by "TV-14".
    *   TV-MA: `1822` titles
    *   TV-14: `1214` titles
    *   R: `778` titles
    *   PG-13: `470` titles
    *   TV-PG: `431` titles

### Release Trends Insights

*   **Number of Titles Released Over Time (by Release Year)**: There's a clear upward trend in content release, peaking in 2017-2018.
    *   2017: `657` titles
    *   2018: `648` titles
    *   2016: `577` titles
    *   2019: `519` titles
    *   2020: `442` titles

### Director & Cast Insights

*   **Directors with the Most Titles**: Jan Suter and Raúl Campos are the most prolific directors in this dataset.
    *   Jan Suter: `21` titles
    *   Raúl Campos: `19` titles
    *   Marcus Raboy: `15` titles
    *   Jay Karas: `15` titles
    *   Cathy Garcia-Molina: `13` titles
*   **Cast Members with the Most Appearances**: Anupam Kher has the highest number of appearances.
    *   Anupam Kher: `42` appearances
    *   Shah Rukh Khan: `34` appearances
    *   Naseeruddin Shah: `31` appearances
    *   Akshay Kumar: `30` appearances
    *   Om Puri: `30` appearances

### Most Frequent Genre Combinations

*   The analysis of `listed_in` after splitting by commas and stripping whitespace shows:
    *   International Movies: `2369`
    *   Dramas: `2293`
    *   Comedies: `1553`
    *   Action & Adventure: `806`
    *   Independent Movies: `740`

### Overall Insights

This analysis provides a comprehensive overview of the Netflix content library, highlighting its strong focus on international movies and dramas, a significant increase in content production in recent years, and a diverse range of content from various countries and for different audience ratings.